frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -1239,4 +1267,117 @@ private void updateVulnerabilityVerification(ResourceRequest request, ResourceRe
         PrintWriter writer = response.getWriter();
         writer.write(responseData.toString());
     }
+
+    private void serveComponentList(ResourceRequest request, ResourceResponse response) throws IOException, PortletException {
```

Comment:
Why does that still throw IOException, but you catch some in the end of this function (line 1284) without rethrowing? Is in case of error a proper 500 returned or is some 2xx returned without content? I do not know the liferay default the behaviour here.

Reply:
removed `IOException`.
Add `response.setProperty(ResourceResponse.HTTP_STATUS_CODE, "500");`
removed `IOException` also for `getFilteredComponentList`